drivers/soundwire/slave.c

```diff
@@ -25,11 +26,18 @@ static int sdw_slave_add(struct sdw_bus *bus,
 {
 	struct sdw_slave *slave;
 	int ret;
+	int i;
 
 	slave = kzalloc(sizeof(*slave), GFP_KERNEL);
 	if (!slave)
 		return -ENOMEM;
 
+	slave->port_ready = kcalloc(SDW_MAX_PORTS, sizeof(*slave->port_ready), GFP_KERNEL);
```

Comment:
then we could go a step further and embed the array into `struct sdw_slave`?

Reply:
yes it's true. Will make that change, thanks for the suggestion.

Reply:
done